File: backend/components/metiers/administration_publique/plugins/samples/sample_plugin.test.py
"""
sample_plugin.test.py
Tests unitaires pour SamplePlugin (clé en main, CI/CD ready)
"""
import pytest
from .sample_plugin import SamplePlugin

def test_sample_plugin_init():
    plugin = SamplePlugin(options={"mode": "test"})
    assert plugin.options == {"mode": "test"}

def test_sample_plugin_config():
    plugin = SamplePlugin()
    assert plugin.init({"level": 1}) is True
    assert plugin.config == {"level": 1}

def test_sample_plugin_run():
    plugin = SamplePlugin()
    plugin.init({"level": 2})
    result = plugin.run("DATA")
    assert result == {"processed": True, "data": "DATA", "config": {"level": 2}}

File: backend/components/metiers/administration_publique/plugins/samples/sample_plugin.test.js
// sample_plugin.test.js
// Tests unitaires pour SamplePlugin (clé en main, CI/CD ready)

const SamplePlugin = require('./sample_plugin');

describe('SamplePlugin', () => {
  it('doit s\'initialiser correctement', () => {
    const plugin = new SamplePlugin({ mode: 'test' });
    expect(plugin.options).toEqual({ mode: 'test' });
  });

  it('doit initialiser la config', () => {
    const plugin = new SamplePlugin();
    expect(plugin.init({ level: 1 })).toBe(true);
    expect(plugin.config).toEqual({ level: 1 });
  });

  it('doit exécuter le run sample', () => {
    const plugin = new SamplePlugin();
    plugin.init({ level: 2 });
    const result = plugin.run('DATA');
    expect(result).toEqual({ processed: true, data: 'DATA', config: { level: 2 } });
  });
});
